### PR TITLE
Feature/timeinvestedapi

### PR DESCRIPTION
**Pull request by SridharJagannathan**
This PR is based on jefferis/elmr#42
I have tested the functionality with the python version of pymaid.get_time_invested
@mmc46 : Can you please suggest any additional changes in the functionality you need

### CODE REVIEW

**Comment by mmc46** [comment]
@SridharJagannathan 
On mode 'ACTIONS', is every action counted, or does the time and number of actions binning still apply? The output shows users with less than 10 actions, so I assume not. Is that correct? If so, it would be good to make it clearer in the description.

Other than that, it seems fine by me.
The other functionality that would be useful is to be able to calculate stats for subsetted neurons, so that we can isolate relevant arbours. 
Subsetting would usually be done via elmr::distal_to() using node tags (see examples there) or nat::pointsinside() for meshes (usually neuropil meshes).

File: NAMESPACE [hunk 1]
```diff
@@ -70,11 +71,13 @@ importFrom(dplyr,right_join)
 importFrom(grDevices,col2rgb)
 importFrom(grDevices,rgb)
 importFrom(jsonlite,fromJSON)
+importFrom(magrittr,"%>%")
 importFrom(nabor,knn)
 importFrom(plyr,rbind.fill)
 importFrom(rgl,as.mesh3d)
 importFrom(rgl,plot3d)
 importFrom(rgl,points3d)
+importFrom(rlang,.data)
```
Comment (jefferis) [inline NAMESPACE]: What is this for?

File: NAMESPACE [hunk 2]
```diff
@@ -70,11 +71,13 @@ importFrom(dplyr,right_join)
 importFrom(grDevices,col2rgb)
 importFrom(grDevices,rgb)
 importFrom(jsonlite,fromJSON)
+importFrom(magrittr,"%>%")
```
Comment (jefferis) [inline NAMESPACE]: The pipe is also exported from `dplyr` (which is already a direct dependency) but in fact `dplyr` gets it from `magrittr` so I don't mind.

File: NAMESPACE (hunk 1 above)
Comment (jefferis) [reply NAMESPACE]: OK I see https://cran.r-project.org/web/packages/dplyr/vignettes/programming.html#different-data-sets. I haven't used that before. `rlang` is also a `dplyr` dependency so no actual increase.

**Comment by jefferis** [comment]
@mmc46 re this:
> The other functionality that would be useful is to be able to calculate stats for subsetted neurons, so that we can isolate relevant arbours.
Subsetting would usually be done via elmr::distal_to() using node tags (see examples there) or nat::pointsinside() for meshes (usually neuropil meshes).

I guess this would require accepting neuronlists as input and then getting just the nodes mentioned in those neurons? That sounds like additional functionality but maybe we could merge the current state into master.

**Comment by mmc46** [comment]
> I guess this would require accepting neuronlists as input and then getting just the nodes mentioned in those neurons?
Yes, I think so.

> That sounds like additional functionality but maybe we could merge the current state into master.
Sounds fine by me.

**Comment by SridharJagannathan** [comment]
@mmc46 
1) I have updated the description in the API to reflect that for mode 'ACTIONS', every action is counted,
irrespective of the parameters like `minimum_actions` and `max_inactive_time`
2) Yes, it would be good to have this functionality may be in a different feature request.
@jefferis 
Can you merge this once everything is Ok?

**Pull request merged**